### DISCUSSION TITLE
Use Ally package in a API Server

### DISCUSSION THREAD

**Discussion by ArthurDanjou** [adonisjs/core#2946] [Help]
I am developing a new section for my personal portfolio and I need to use Adonisjs as a API server. I added the ally package in its latest version to simplify the connection of the users. I use Discord, Twitter, Google and GitHub drivers. I followed the documentation and added the redirect route and the callback route. All is working as I want if I access the GET route from the browser or from Postman. 

My portfolio is a nuxt application. I use axioms to do the same GET request but I have a issue with the CORS. 

You can check the source here :
- Website: https://github.com/ArthurDanjou/artsite/tree/dev
- Api server : https://github.com/ArthurDanjou/artapi/tree/dev 

The files are guestbook.vue (NUXT) and AuthController.ts (ADONIS)

**Comment by thetutlage** [comment]
Do you expect someone to scan your entire codebase and come up with solution for you? Just share the relevant code snippets or link to them

**Reply by ArthurDanjou** [reply]
Hello, sorry for the late response. 

You can find my [vue component](https://github.com/ArthurDanjou/artsite/blob/dev/src/components/GuestBookForm.vue) where I do my GET request to my API to use the `ally` package.
You can see my [adonis controller](https://github.com/ArthurDanjou/artapi/blob/dev/app/Controllers/Http/AuthController.ts).
Here is my [cors config](https://github.com/ArthurDanjou/artapi/blob/dev/config/cors.ts).

I have the following error when I try to reach my API : `No 'Access-Control-Allow-Origin' header is present on the requested resource`. 

I use Traefik as a reverse proxy, I dont know if it can remove or clear the headers of the requests and the responses.

Thank you for your help

**Comment by ArthurDanjou** [comment]
My current Workflow to integrate Oauth in my case is : ![Oauth scheme](https://user-images.githubusercontent.com/29738535/128224786-70b762d4-0489-4529-8558-ef7ad2d28d0e.png).
Is it correct ?

**Reply by ArthurDanjou** [reply]
In this workflow, by following the adonisjs/ally documentation, all is working if I start from the 2nd step and finish to the 3rd step. I do a GET request (currently `/auth/github`) and Adonisjs returns well my user. But if I try to reach the same route from my Nuxt app, I have a CORS problem

**Reply by McSneaky** [reply]
> But if I try to reach the same route from my Nuxt app, I have a CORS problem

Is it working for you locally or you get CORS errors only in prod?


> My current Workflow to integrate Oauth in my case is .. Is it correct ?

That's correct indeed, that's how it should work

**Reply by ArthurDanjou** [reply]
It isn't working in development in local. In the cors config, as you can see [here](https://github.com/ArthurDanjou/Athena/blob/dev/config/cors.ts), the origin section accepts all origin in dev mode but I have a missing cross origin header as mentioned previously

**Reply by ArthurDanjou** [reply]
If you have a public project where I can see your Adonis app, I'll take it. In addition, I use a reverse proxy (traefik). Does it clear the headers ?
